indexing-service/src/main/java/org/apache/druid/indexing/overlord/ZkWorker.java

```diff
@@ -63,6 +66,27 @@ public ZkWorker(Worker worker, PathChildrenCache statusCache, final ObjectMapper
     this.statusCache = statusCache;
     this.cacheConverter = (ChildData input) ->
         JacksonUtils.readValue(jsonMapper, input.getData(), TaskAnnouncement.class);
+    this.taskIdExtractor = createTaskIdExtractor(jsonMapper);
+  }
+
+  static java.util.function.Function<ChildData, String> createTaskIdExtractor(final ObjectMapper jsonMapper)
+  {
+    return (ChildData input) -> {
+      try {
+        JsonNode parent = jsonMapper.readTree(input.getData());
```

Comment:
`readTree` still deserializes all the input to object. Since here we just want to get the value of `id` field, I think we can use jackson's streaming parser api to get the id so that we don't need to parse all the string content.

Reply:
You're totally right. Let me work on this. :facepalm: ..

Reply:
OK, captured this in d5a9864c8d8f79e7b8235321d9f33402206f568d. Thanks for feedback.

Reply:
FWIW this fix done properly converts this (the block on the right-hand side) ...

![image](https://user-images.githubusercontent.com/3196528/147312284-73bdd9af-72e2-4c5c-99c1-7db8d7c7b482.png)

to this ...

![image](https://user-images.githubusercontent.com/3196528/147312309-373def60-3f79-4121-b9ae-47b9ffa68e9f.png)